### PR TITLE
chore: Improve the release pipeline

### PR DESCRIPTION
**Pull request by zepfred**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -121,4 +120,28 @@ jobs:
...
+      - name: Update release branch
+        working-directory: ./timefold-quickstarts
+        shell: bash
+        run: |
+          java_version=${{ github.event.inputs.javaVersion }}
+          version="${java_version%.*}.x"
+          echo $version
+          exists="$(git branch -a | grep -w $version || true)"
+          echo "branch $exists"
+          if [ -n "$exists" ]; then
```
Comment (triceo) [inline .github/workflows/release.yml]: Why are we doing this if none of workflows in the other repos had this?
Can't we delete the branch if it exists, like we do everywhere else? For consistency.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Release
+'name: Release
```
Comment (triceo) [inline .github/workflows/release.yml]: ?

**Review by triceo — CHANGES_REQUESTED** [review]
I'm not sure this file is valid, or I don't understand the `'` syntax.
Also, we switched to a new way of publishing Maven Central artifacts, and so the release pipeline will have to be updated more.

Let's discuss.

File: .github/workflows/release.yml (hunk 2 above)
Comment (zepfred) [reply .github/workflows/release.yml]: This is a typo

**Comment by zepfred** [comment]
> I'm not sure this file is valid, or I don't understand the `'` syntax. Also, we switched to a new way of publishing Maven Central artifacts, and so the release pipeline will have to be updated more.
> 
> Let's discuss.

We create a branch named `release_branch_...`, which contains changes related to the release version. Next, we will create a pull request and merge all changes from the release branch into the stable branch. The final three steps are to move the release branch back to `999-dev`, check if the version branch (`v1.16.x`) exists, and rename the release branch to the version branch instead of deleting it.

**Review by triceo — APPROVED** [review]

**Pull request merged**